thunder/tests/test_transforms.py

```diff
@@ -221,3 +221,56 @@ def test_nvfuser_cse():
         assert prologue_proxy.device == thunder.core.devices.to_device(t.device)
         assert comp_proxy.dtype == thunder.core.dtypes.to_dtype(t.dtype)
         assert prologue_proxy.dtype == thunder.core.dtypes.to_dtype(t.dtype)
+
+
+@requiresCUDA
+def test_materialization_init():
+    from thunder.transforms import MaterializationTransform
+    from thunder.transforms.quantization import BitsAndBytesLinearQuant4bit, get_bitsandbytes_executor
+
+    bitsandbytes_executor = get_bitsandbytes_executor()
+
+    def get_model():
+        m0 = torch.nn.Linear(2, 2)
+
+        # to not change the rng state
+        with torch.device("meta"):
+            m4 = torch.nn.Linear(2, 2)
```

Comment:
with repeated `torch.manual_seed` below, I'm not quite following whether this comment / this use of meta device matter...

Reply:
This is the RNG state within one module initialization (i.e. if this called randn for device tensors, it would put the parameters here that would usually be in the middle layer, the `manual_seed` only sets the seed).
Note that this is part of #980 and just shows here due to a missing rebase.